code/modules/mapping/random_rooms.dm

```diff
@@ -1478,3 +1478,41 @@
 	centerspawner = FALSE
 	template_height = 11
 	template_width = 5
+
+
+/datum/map_template/random_room/engines/box/supermatter
```

Comment:
/datum/map_template/random_room/engines
/datum/map_template/random_room/engines/box/
and /datum/map_template/random_room/engines/meta
These need to have defined names, or the typepath needs to be rewritten.
It creates errors on load because their subtypes do not have anything defined, only their children.
I'd just suggest making it '/datum/map_template/random_room/box_supermatter' and such.